lib/fluent/plugin/kafka_producer_ext.rb

```diff
@@ -243,14 +245,14 @@ def deliver_messages_with_retries
 
     def assign_partitions!
       failed_messages = []
-      partition_count = @cluster.partitions_for(@topic).count
 
       @pending_message_queue.each do |message|
         partition = message.partition
 
         begin
           if partition.nil?
-            partition = Partitioner.call(partition_count, message)
+            partition_count = @cluster.partitions_for(@topic).count
```

Comment:
Why move this line?
In `TopicProducer`, `@topic` is fixed value so no need to call for each message.

Reply:
Move the count back (out of the loop).